external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/ManualPartitionSubscription.java

```diff
@@ -58,21 +51,21 @@ public ManualPartitionNamedSubscription(ManualPartitioner parter, String ... top
     
     @Override
     public void refreshAssignment() {
-        List<TopicPartition> allPartitions = new ArrayList<>();
-        for (String topic : topics) {
-            for (PartitionInfo partitionInfo: consumer.partitionsFor(topic)) {
-                allPartitions.add(new TopicPartition(partitionInfo.topic(), partitionInfo.partition()));
-            }
-        }
+        List<TopicPartition> allPartitions = partitionFilter.getFilteredTopicPartitions(consumer);
         Collections.sort(allPartitions, TopicPartitionComparator.INSTANCE);
         Set<TopicPartition> newAssignment = new HashSet<>(partitioner.partition(allPartitions, context));
         if (!newAssignment.equals(currentAssignment)) {
+            consumer.assign(newAssignment);
             if (currentAssignment != null) {
                 listener.onPartitionsRevoked(currentAssignment);
-                listener.onPartitionsAssigned(newAssignment);
             }
+            listener.onPartitionsAssigned(newAssignment);
             currentAssignment = newAssignment;
-            consumer.assign(currentAssignment);
```

Comment:
@srdo lines 62 and 63 should be switched. We want to call the listener only once all is set.